eng/tools/spec-gen-sdk-runner/src/change-files.ts

```diff
@@ -0,0 +1,96 @@
+import path from "node:path";
+import {
+  execAsync,
+  getChangedFiles,
+  searchRelatedParentFolders,
+  searchRelatedTypeSpecProjectBySharedLibrary,
+  searchSharedLibrary,
+} from "./utils.js";
+import { logMessage } from "./log.js";
+import { SpecGenSdkCmdInput } from "./types.js";
+
+const readmeMdRegex = /^readme.md$/;
+const typespecProjectRegex = /^tspconfig.yaml$/;
+const typespecProjectSharedLibraryRegex = /[^/]+\.Shared/;
+
+type ChangedSpecs = {
+  [K in "readmeMd" | "typespecProject"]?: string;
+} & {
+  specs: string[];
+};
+
+export async function detectChangedSpecConfigFiles(
+  commandInput: SpecGenSdkCmdInput,
+): Promise<ChangedSpecs[]> {
+  const prChangedFiles: string[] = getChangedFiles(commandInput.localSpecRepoPath) ?? [];
+  if (prChangedFiles.length === 0) {
+    logMessage("No files changed in the PR");
+  }
+  logMessage(`Changed files in the PR: ${prChangedFiles.length}`);
+  const fileList = prChangedFiles.filter((p) => !p.includes("/scenarios/"));
+  const { stdout: headCommitRaw } = await execAsync("git rev-parse HEAD");
+  const headCommit = headCommitRaw.trim(); // Trim any newline characters
+  const { stdout: treeIdRaw } = await execAsync(`git rev-parse ${headCommit}^{tree}`);
+  const treeId = treeIdRaw.trim();
```

Comment:
Why are we looking at git here? We should just be using the files on disk and assume that repo is correctly checked out.